### PR TITLE
make verify-manifests

### LINKED ISSUE #236: Add verify-manifests make target and CI job
**What**

Add a verify-manifests make target than executes `make manifests` and returns a non zero exit code if anything chnages in the config directory. Run this is part of our PR CI jobs.

Take inspiration from https://github.com/Kuadrant/kuadrant-operator/blob/main/make/verify.mk#L7 https://github.com/Kuadrant/kuadrant-operator/blob/main/.github/workflows/test.yaml#L79

**Why**

Catch issue like https://github.com/Kuadrant/multicluster-gateway-controller/pull/235#discussion_r1221034170

### PR DESCRIPTION
**Pull request by jasonmadigan**
- adds a new `verify-manifests` target to the makefile
- added this to our GH Action CI

fixes #236

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -88,11 +88,11 @@ ifndef ignore-not-found
 endif
 
 .PHONY: install
-install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
+install: verify-manifests manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
```
Comment (jasonmadigan) [inline Makefile:91]: perhaps not needed, given the action will run - but it may be better to warn about this in multiple targets?

**Comment by jasonmadigan** [comment]
/hold

Until latest `config/rbac/role.yaml` checked in

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -33,6 +33,20 @@ jobs:
...
           args: --timeout=5m
           skip-go-installation: true
+  verify-manifests:
+    name: Verify manifests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.18.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18.x
```
Comment (mikenairn) [inline .github/workflows/ci.yaml:43]: Probbaly want to use v1.19

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -33,6 +33,20 @@ jobs:
           only-new-issues: true
           args: --timeout=5m
           skip-go-installation: true
+  verify-manifests:
+    name: Verify manifests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.18.x
```
Comment (mikenairn) [inline .github/workflows/ci.yaml:40]: 1.19

File: Makefile (hunk 1 above)
Comment (mikenairn) [reply Makefile]: Yeah, this would fail locally until you committed the changes in which would be annoying

**Comment by mikenairn** [comment]
/lgtm
/approve

I'm not sure if the failing test will stop it merging with the current setup, hopefully it does, but the hold will prevent it merging anyway.

**Comment by mikenairn** [comment]
We cold just commit this change as part of this PR https://github.com/Kuadrant/multicluster-gateway-controller/pull/235/files#diff-6afa7d73841a5c0ebdb78eaf859ce5a400203a037f871440940b5347c3a97376L95.

**Comment by jasonmadigan** [comment]
Might be worth adding that just to test the CI then passes.

**Pull request merged**